x-pack/platform/packages/shared/kbn-classic-stream-flyout/src/components/create_classic_stream_flyout/steps/name_and_confirm/name_stream_section.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiPanel, EuiSpacer, EuiTitle, EuiSelect, useEuiTheme } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import type { ValidationErrorType } from '../../../../utils';
+import { StreamNameInput } from '../../../stream_name_input';
+
+const getValidationErrorMessage = (
+  validationError: ValidationErrorType,
+  conflictingIndexPattern?: string
+) => {
+  if (validationError === 'empty') {
+    return i18n.translate(
+      'xpack.createClassicStreamFlyout.nameAndConfirmStep.emptyValidationError',
+      {
+        defaultMessage:
+          'Please supply a valid text string for all wildcards within the selected index pattern.',
+      }
+    );
+  }
+
+  if (validationError === 'duplicate') {
+    return i18n.translate(
+      'xpack.createClassicStreamFlyout.nameAndConfirmStep.duplicateValidationError',
+      {
+        defaultMessage: 'This stream name already exists. Please try a different name.',
+      }
+    );
+  }
+
+  if (validationError === 'higherPriority' && conflictingIndexPattern) {
+    return [
+      <FormattedMessage
+        key="higherPriorityError"
+        id="xpack.createClassicStreamFlyout.nameAndConfirmStep.higherPriorityValidationError"
+        defaultMessage="This stream name matches a higher priority index template with index pattern {pattern}. Please alter your stream name so that it no longer conflicts, or consider changing your selected index template."
+        values={{
+          pattern: <strong>{conflictingIndexPattern}</strong>,
+        }}
+      />,
+    ];
+  }
+
+  return undefined;
+};
+
+interface NameStreamSectionProps {
+  indexPatterns: string[];
+  selectedIndexPattern: string;
+  onIndexPatternChange: (pattern: string) => void;
+  onStreamNameChange: (streamName: string) => void;
+  validationError: ValidationErrorType;
+  conflictingIndexPattern?: string;
+}
```

Comment:
These types were already defined in `name_and_confirm/name_and_confirm_step.tsx` any chance we can refactor them into a types.ts file and then consume from both places? If we need an extra prop we can always create a new interface and extend the base one